### ISSUE TITLE
ProjectRoots not being removed when build on Linux and hosted on Windows

### ISSUE THREAD

**Issue by sgtfrankieboy** [bugsnag/bugsnag-dotnet#152]
### Describe the bug
When a project is build on Linux but hosted on Windows, the defined ProjectRoots are not being removed.

This is being caused by:
https://github.com/bugsnag/bugsnag-dotnet/blob/b4d91276f1e66d91381d37cb16fef4d8c7992d6b/src/Bugsnag/Middleware.cs#L40-L48

It checks if the last character of the ProjectRoot is equal to `DirectorySeparatorChar`. Because the project is build on Linux the final character in the root folder is `/` but its being ran on Windows so it checks for `\` instead.

Any reason why that particular check exists in the first place? The quickest solution would be to just remove it. Although if there is a reason for it, it should be altered so it checks for both `/` and `\` plus whatever the default separator character is.

### Environment
* Bugsnag version: 2.2.3
* .NET framework version: .NET 6

**Comment by johnkiely1** [comment]
Hi @sgtfrankieboy, 

Would you mind confirming how you are defining your projects roots in the Bugsnag configuration?

**Comment by sgtfrankieboy** [comment]
We setup the projects roots like this:

```
bugsnagConfig.ProjectRoots = new[]
{
    "/builds/project/src/backend/",
};
```

**Comment by johnkiely1** [comment]
Thanks, we're going to take a look at this and see what we can do to resolve it.

**Comment by yousif-bugsnag** [comment]
This is now fixed in [v3.0.1](https://github.com/bugsnag/bugsnag-dotnet/releases/tag/v3.0.1) 🎉
